src/bootstrap/tool.rs

```diff
@@ -162,13 +162,16 @@ impl Step for ToolBuild {
                 "the following dependencies are duplicated although they \
                       have the same features enabled:"
             );
-            for (id, cur, prev) in duplicates.drain_filter(|(_, cur, prev)| cur.2 == prev.2) {
+            duplicates.retain(|(_, cur, prev)| cur.2 != prev.2);
+            let (same, different): (Vec<_>, Vec<_>) =
+                duplicates.into_iter().partition(|(_, cur, prev)| cur.2 == prev.2);
```

Comment:
You already removed the ones that are the same on line 165. So they're gone now, `same` will be empty. 
```suggestion
            let (same, different): (Vec<_>, Vec<_>) =
                duplicates.into_iter().partition(|(_, cur, prev)| cur.2 == prev.2);
```